stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+    >>> counts, _ = np.histogram(times, bins=np.linspace(0, 1000, 11))
+    >>> bin_times = np.arange(50, 1000, 100)
+    >>> get_total_ctrate(bin_times, gti, 1000, counts=counts)
+    1.0
+    >>> get_total_ctrate(times, gti, 1000)
+    1.0
+    """
+    Nph = 0
+    Nintvs = 0
+    func = _which_segment_idx_fun(counts)
+
+    for _, _, idx0, idx1 in func(times, gti, segment_size):
+        if counts is None:
+            Nph += idx1 - idx0
+        else:
+            Nph += np.sum(counts[idx0:idx1])
+        Nintvs += 1
+
+    return Nph / (Nintvs * segment_size)
+
+
+def get_flux_iterable_from_segments(times, gti, segment_size, N=None, counts=None, errors=None):
+    """Get fluxes from different segments of the observation.
+
+    If ``counts`` is ``None``, the input times are interpreted as events.
+    At least one between ``N`` and ``counts`` needs to be specified.
+    Otherwise, they are assumed uniformly binned inside each GTI, and the number
+    of events per bin is taken from ``counts``
+
+    Parameters
+    ----------
+    times : float `np.array`
+        Array of times
+    gti : [[gti00, gti01], [gti10, gti11], ...]
+        good time intervals
+    segment_size : float
+        length of segments
+
+    Other parameters
+    ----------------
+    N : int, default None
+        Number of bins to divide the ``segment_size`` in
+    counts : float `np.array`, default None
+        Array of counts per bin
+
+    Returns
+    -------
+    cts : `np.array`
+        Array of counts
+    err : `np.array`
+        (optional) if ``errors`` is None, an array of errors in the segment
+
+    """
+    if counts is None and N is None:
+        raise ValueError(
+            "At least one between counts (if light curve) and N (if events) has to be set"
+        )
+
+    dt = None
+    if counts is not None:
+        dt = np.median(np.diff(times[:100]))
+
+    fun = _which_segment_idx_fun(counts, dt)
+
+    for s, e, idx0, idx1 in fun(times, gti, segment_size):
+        if idx1 - idx0 < 2:
+            yield None
+            continue
+        if counts is None:
+            event_times = times[idx0:idx1]
+            # counts, _ = np.histogram(event_times - s, bins=bins)
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = histogram((event_times - s).astype(float), bins=N,
+                            range=[0, segment_size]).astype(float)
+            cts = np.array(cts)
+        else:
+            cts = counts[idx0:idx1].astype(float)
+            if errors is not None:
+                cts = cts, errors[idx0:idx1]
+
+        yield cts
+
+
+def avg_pds_from_iterable(flux_iterable, dt, norm="abs", use_common_mean=True, silent=False):
+    """Calculate the average periodogram from an iterable of light curves
+
+    Parameters
+    ----------
+    flux_iterable : `iterable` of `np.array`s or of tuples (`np.array`, `np.array`)
```

Comment:
When you say "iterable", do you actually mean that? Because if you stick in here what comes out of the preceding function, that would be a generator? That might be an important distinction to make.

Reply:
Any iterable object. It can be a list, a generator, a tuple...